### DISCUSSION TITLE
Metadata classes do not implement toString

### DISCUSSION THREAD

**Discussion by bulldog98** [smallrye/smallrye-reactive-messaging#2224] [Q&A] [answered]
I tried logging the metadata of a message and noticed that the `org.eclipse.microprofile.reactive.messaging.Metadata` class does not implement a reasonable toString method.

I tried working around to no toString on the `Metadata` as follows:
```java
    @Incoming("some-channel")
    public CompletionStage<Void> process(final @NotNull Message<byte[]> event) {
        final var payload = event.getPayload();
        final var list = new ArrayList<>();
        event.getMetadata().iterator().forEachRemaining(list::add);
        log.info("some-channel: " + Arrays.toString(payload) + " " + list);
        return event.ack();
    }
```
And got a log like this `some-channel: [10] [io.smallrye.reactive.messaging.rabbitmq.IncomingRabbitMQMetadata@271b214f, org.eclipse.microprofile.reactive.messaging.Metadata@13b7f3fb, io.smallrye.reactive.messaging.providers.locals.LocalContextMetadata@5c4e7740]`.

My Question is it intended that logging Metadata is difficult?

**Comment by cescoffier** [comment, marked as answer]
Metadata is just a container. The contained object may not have an understandable `toString`. 
The best we could do is to list the included types - but if you have a metadata with a nice `toString`, it would not be used.

For example, in your case it would produce: [IncomingRabbitMQMetadata, Metadata, LocalContextMetadata] (with the FQCN).
So, I'm not sure of its usefulness.
